src/fixtures/base_fixtures.ts

```diff
@@ -0,0 +1,60 @@
+import { test as base} from '@playwright/test';
+import BasePage from '../../pages/BasePage';
+import BookStorePage from '../../pages/BookStorePage';
+import ButtonsPage from '../../pages/ButtonsPage';
+import ElementsPage from '../../pages/ElementsPage';
+import LoginPage from '../../pages/LoginPage';
+import RegistrationPage from '../../pages/RegistrationPage';
+
+type MyFixture = {
+    expectedCategoryArray: string[],
+    login: LoginPage;
+    elementsPage: ElementsPage;
+    buttonsPage: ButtonsPage;
+    basePage: BasePage;
+    registrationPage: RegistrationPage;
+    bookStorePage: BookStorePage;
+}
+
+const test = base.extend <MyFixture>  ({
+    expectedCategoryArray: [
+        "Elements",
+        "Forms",
+        "Alerts, Frame & Windows",
+        "Widgets",
+        "Interactions",
+        "Book Store Application" 
+    ],
+    login: async({page}, use) => {
+        const login = new LoginPage(page);
+        await login.clickBookStoreApplication();
+        await login.clickLoginSideBar();
+        await use(login);
+        },
+    elementsPage: async({page}, use) => {
+        const elementsPage = new ElementsPage(page);
+        await use(elementsPage);
```

Comment:
keep a shorter format, in case you need initialize a class
```suggestion
         await use(new ElementsPage(page));
```